src/bsoncxx/types.hpp

```diff
@@ -346,10 +346,6 @@ BSONCXX_INLINE bool operator==(const b_date& lhs, const b_date& rhs) {
 ///
 /// A BSON null value.
 ///
-/// @warning
-///   This BSON type is used internally by the MongoDB server - use by clients
-///   is discouraged.
-///
```

Comment:
To clarify - I think we should also remove the warning for the places were it is improperly applied.

The deprecation thing is going to be tricky. We need to think of a strategy that:

- Allows us to deprecate a class/method/enum/symbol for consumers of the library
- Allows us to continue to use the symbol internally while using -Wall -Werror
- Does not break ABI.

The traditional way to do this, without regard to ABI, is to rename the (lets say class X) to X_no_deprecated, then using it in as X with a deprecation attribute. Then our code, when it needs to use X, uses X_no_deprecated so that it doesn't cause deprecation warnings when we use it (or, when a user of the library explicitly decides to sidestep the deprecation warning). However, doing that is an ABI breaking change, since now any methods of X have moved to the method X_no_deprecated. So, we can't do that. Something we will need to think about during 3.1 development as we move towards stabilizing ABI.